### PR TITLE
ES-2345: corda runtime gradle plugin test pipeline

### PR DESCRIPTION
**Pull request by anton-subbotin**
- Add basic tests to verify the pipeline and the env are operational:
  - Lifecycle test:
    1. Run startCorda task, which starts docker compose project of Kafka-enabled Combined Worker.
    2. Wait until Corda is launched, and run stop task
  - Queries test class shows example of test env setup by starting and stopping Corda using docker compose in the beforeAll/afterAll
- Corda image versions: get from environment (from the pipeline), or use a hardcoded stable version as a fallback. This is to run tests in the pipeline using Corda of the same revision as the plugin.
- Add pipeline for the task `cordaRuntimeGradlePluginSmokeTest`: https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-gradle-plugin-smoke-tests/view/change-requests/job/PR-6209/27/
  - Shared pipeline PR: https://github.com/corda/corda-shared-build-pipeline-steps/pull/2043

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ cordaPipelineKubernetesAgent(
     e2eTestName: 'corda-runtime-os-e2e-tests',
     runE2eTests: true,
     combinedWorkere2eTests: true,
+    runtimeGradlePluginTests: true,
```
Comment (tlawson3) [inline Jenkinsfile:13]: So does this mean any PR in runtime-os will run the gradle plugin tests?

File: tools/corda-runtime-gradle-plugin/src/endToEndTest/kotlin/net/corda/gradle/plugin/EndToEndTestBase.kt [hunk 2]
```diff
@@ -1,26 +1,95 @@
...
+import java.net.URI
+import java.time.Duration
+import java.time.Instant
+import java.util.concurrent.TimeUnit
 
 // https://docs.gradle.org/current/userguide/test_kit.html
 abstract class EndToEndTestBase {
-    // TODO based on integrationTest/kotlin/net/corda/gradle/plugin/FunctionalBaseTest.kt
-    //  for now contains only necessary methods and properties
+    companion object {
```
Comment (tlawson3) [inline tools/corda-runtime-gradle-plugin/src/endToEndTest/kotlin/net/corda/gradle/plugin/EndToEndTestBase.kt]: The inspiration for this work was the corda-runtime-os-combined-worker-plugins-smoke-tests so how about we call these CordaRuntimeGradlePluginSmokeTests

File: Jenkinsfile (hunk 1 above)
Comment (anton-subbotin) [reply Jenkinsfile:13]: yes. It's one of the parallel stages so it won't affect PR gate times.

File: tools/corda-runtime-gradle-plugin/src/endToEndTest/kotlin/net/corda/gradle/plugin/EndToEndTestBase.kt (hunk 2 above)
Comment (anton-subbotin) [reply tools/corda-runtime-gradle-plugin/src/endToEndTest/kotlin/net/corda/gradle/plugin/EndToEndTestBase.kt]: renamed source sets and dir to `smokeTest` to be short; Gradle task is named `cordaRuntimeGradlePluginSmokeTest`. 

The task is _not_ included if just `./gradlew :smokeTest` is run, as expected.

**Review by tlawson3 — DISMISSED** [review]

File: .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+]
+
+GitUtils gitUtils = new GitUtils(this)
+KubernetesAgent k8s = new KubernetesAgent(
+        BuildEnvironment.AMD64_LINUX_JAVA17,
+        KubernetesCluster.JenkinsAgents,
+        3
+).withDocker()
+
+pipeline {
```
Comment (tlawson3) [inline .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent:27]: What was the reason you've moved this out of the shared pipeline and into here?

Comment (ronanbrowne) [reply .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent:27]: See my comments in shared lib PR

Comment (ronanbrowne) [reply .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent:27]: https://github.com/corda/corda-shared-build-pipeline-steps/pull/2043#discussion_r1644942061

File: .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent [hunk 4]
```diff
@@ -0,0 +1,120 @@
+@Library('corda-shared-build-pipeline-steps@5.3') _
+
+import com.r3.build.agents.KubernetesAgent
+import com.r3.build.enums.BuildEnvironment
+import com.r3.build.enums.KubernetesCluster
+import com.r3.build.utils.GitUtils
+
+Map config = [
```
Comment (ronanbrowne) [inline .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent:8]: config map can be removed now as no need for overrides. Just set the properties as needed in this file

Comment (anton-subbotin) [reply .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent:8]: I'd rather keep it on top of the file for cleaner configurability, if necessary for debug/replay etc.

**Review by tlawson3 — APPROVED** [review]

File: .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent (hunk 4 above)
Comment (ronanbrowne) [reply .ci/JenkinsfileCordaRuntimeGradlePluginTestsKubernetesAgent:8]: Fair enough

**Review by ronanbrowne — APPROVED** [review]
LGTM - pending clean run of PR gates

**Comment by anton-subbotin** [comment]
build e2e

**Pull request merged**